prusti-common/src/vir/optimizations/functions/simplifier.rs

```diff
@@ -46,134 +50,118 @@ impl ExprSimplifier {
                     box ast::Expr::Const(ast::ConstExpr {
                         value: ast::Const::Bool(b),
-                        ..
+                        position: inner_pos,
                     }),
                 position: pos,
             }) => ast::Expr::Const(ast::ConstExpr {
                 value: ast::Const::Bool(!b),
                 position: pos,
-            }),
+            })
+            .set_default_pos(inner_pos),
             ast::Expr::UnaryOp(ast::UnaryOp {
                 op_kind: ast::UnaryOpKind::Not,
                 argument:
                     box ast::Expr::BinOp(ast::BinOp {
                         op_kind: ast::BinaryOpKind::EqCmp,
                         box left,
                         box right,
-                        ..
+                        position: inner_pos,
                     }),
                 position: pos,
-            }) => ast::Expr::BinOp(ast::BinOp {
+            }) if !matches!(left.get_type(), ast::Type::Float(_)) => ast::Expr::BinOp(ast::BinOp {
                 op_kind: ast::BinaryOpKind::NeCmp,
                 left: box left,
                 right: box right,
                 position: pos,
-            }),
-            ast::Expr::BinOp(ast::BinOp {
-                op_kind: ast::BinaryOpKind::And,
-                left:
-                    box ast::Expr::Const(ast::ConstExpr {
-                        value: ast::Const::Bool(b1),
-                        ..
-                    }),
-                right:
-                    box ast::Expr::Const(ast::ConstExpr {
-                        value: ast::Const::Bool(b2),
-                        ..
-                    }),
-                position: pos,
-            }) => ast::Expr::Const(ast::ConstExpr {
-                value: ast::Const::Bool(b1 && b2),
-                position: pos,
-            }),
+            })
+            .set_default_pos(inner_pos),
             ast::Expr::BinOp(ast::BinOp {
                 op_kind: ast::BinaryOpKind::And,
                 left:
                     box ast::Expr::Const(ast::ConstExpr {
                         value: ast::Const::Bool(b),
-                        ..
+                        position: inner_pos,
                     }),
                 right: box conjunct,
-                ..
+                position: pos,
             })
             | ast::Expr::BinOp(ast::BinOp {
                 op_kind: ast::BinaryOpKind::And,
                 left: box conjunct,
                 right:
                     box ast::Expr::Const(ast::ConstExpr {
                         value: ast::Const::Bool(b),
-                        ..
+                        position: inner_pos,
                     }),
-                ..
-            }) => {
-                if b {
-                    conjunct
-                } else {
-                    false.into()
-                }
+                position: pos,
+            }) => if b {
+                conjunct
```

Comment:
@vakaras If we inline a `dummy` trigger function body (with the body `true`) and also simplify, this would change semantics of quantifiers. Do you know in what order the inlining is performed?